src/ds5/ds5-factory.cpp

```diff
@@ -956,6 +958,43 @@ namespace librealsense
 
             return tags;
         };
+
+        std::shared_ptr<safety_preset> get_safety_preset_at_index(int index) override
+        {
+            std::shared_ptr<safety_preset> result;
+            
+            // prepare command
+            command cmd(ds::SAFETY_PRESET_READ);
+            cmd.require_response = false;
+            cmd.param1 = index;
+
+            // send command to device and get response
+            std::vector< uint8_t > response = ds5_device::_hw_monitor->send(cmd);
+
+            // convert response to safety_preset struct
+            assert(response.size() == sizeof(result));
```

Comment:
assert is only for debug configuration, what will happen on release if we get 0 or other value than the size?